torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1221,8 +1221,8 @@ void check_gpu_single_tensor(const at::Tensor& tensor) {
   if (!tensor.is_cuda() || tensor.is_sparse()) {
     TORCH_CHECK(false, "Tensors must be CUDA and dense");
   }
-  if (!tensor.is_contiguous()) {
-    TORCH_CHECK(false, "Tensors must be contiguous");
+  if (!tensor.is_non_overlapping_and_dense()) {
```

Comment:
Do we have tests covering this change?
We're now asking more of NCCL so we should cover this new condition with additional tests.

Reply:
I don't think the collectives that already support `is_non_overlapping_and_dense` have any test for this case, but can add tests for all of them in this PR.